Orange/widgets/tests/test_settings_handler.py

```diff
@@ -64,6 +64,37 @@ def test_write_defaults(self):
 
         os.remove(settings_file)
 
+    def test_write_PermissionError(self):
+        """
+        Error may occur while trying to open file. Therefore file
+        cannot be closed. Tests if error is handled.
+        GH-2147
+        """
+        _, settings_file = mkstemp(suffix='.ini')
+        handler = SettingsHandler()
+        handler._get_settings_filename = lambda: settings_file  # pylint: disable=W0212
+        from sys import version_info
+        patch_target = "Orange.widgets.settings.open" if version_info >= (3, 5) \
+            else "builtins.open"
+        with patch(patch_target, side_effect=PermissionError):
+            handler.write_defaults()
+
+    def test_write_EOF_IO_Pickling_Errors(self):
+        """
+        Tests if errors are handled.
+        GH-2147
+        """
+        _, settings_file = mkstemp(suffix='.ini')
+        patch_target = "Orange.widgets.settings.SettingsHandler.write_defaults_file"
+        handler = SettingsHandler()
+        handler._get_settings_filename = lambda: settings_file  # pylint: disable=W0212
+        with patch(patch_target, side_effect=EOFError):
+            handler.write_defaults()
+        with patch(patch_target, side_effect=IOError):
+            handler.write_defaults()
+        with patch(patch_target, side_effect=pickle.PicklingError):
+            handler.write_defaults()
```

Comment:
You could also check that the file gets deleted (as it contains invalid settings).